plugins/services/src/js/components/forms/ContainerServiceFormSection.js

```diff
@@ -241,107 +266,82 @@ class ContainerServiceFormSection extends Component {
-        <a href={MetadataStore.buildDocsURI('/usage/tutorials/registry/#docs-article')} target="_blank">
+        <a
+          href={MetadataStore.buildDocsURI('/usage/tutorials/registry/#docs-article')}
+          target="_blank">
           More information
         </a>.
       </span>
     );
 
+    if (containerType == null || containerType === NONE) {
+      tooltipContent = 'Mesos Runtime does not support container images, please select Docker Runtime or Universal Container Runtime if you want to use container images.';
+      iconID = 'lock';
+    }
+
     return (
-      <FieldLabel key="image-label">
+      <FieldLabel>
         {'Container Image '}
         <Tooltip
           content={tooltipContent}
           interactive={true}
           wrapText={true}
           maxWidth={300}
           scrollContainer=".gm-scroll-view">
-          <Icon color="grey" id="circle-question" size="mini" />
+          <Icon color="grey" id={iconID} size="mini" />
         </Tooltip>
       </FieldLabel>
     );
   }
 
-  getImageInput(data) {
-    const {container = {}} = data;
-    const image = findNestedPropertyInObject(container, 'docker.image');
-
-    if (container == null || container.type == null ||
-        container.type === NONE) {
-      return [
-        <FieldLabel key="container-image-input-tooltip">
-          {'Container Image '}
-          <Tooltip
-            content="Mesos Runtime does not support container images, please select Docker Runtime or Universal Container Runtime if you want to use container images."
-            interactive={true}
-            maxWidth={300}
-            scrollContainer=".gm-scroll-view"
-            wrapText={true}>
-            <Icon color="grey" id="lock" size="mini" />
-          </Tooltip>
-        </FieldLabel>,
-        <FieldInput
-          key="container-image-input"
-          name="container.docker.image"
-          disabled={true}
-          value={image} />
-      ];
-    }
-
-    return [
-      this.getImageLabel(),
-      <FieldInput
-        key="image-field"
-        name="container.docker.image"
-        value={image} />,
-      <FieldHelp key="image-help">
-        Enter a Docker image you want to run, e.g. nginx.
-      </FieldHelp>
-    ];
-  }
-
   render() {
-    const {data, errors} = this.props;
-
+    const {data, errors, path} = this.props;
+    const containerType = findNestedPropertyInObject(data, `${path}.type`);
+    const image = findNestedPropertyInObject(data, `${path}.docker.image`);
+    const imageDisabled = containerType == null || containerType === NONE;
     const imageErrors = findNestedPropertyInObject(
       errors,
-      'container.docker.image'
-    );
-
-    const containerType = findNestedPropertyInObject(
-      data,
-      'container.type'
+      `${path}.docker.image`
     );
 
     return (
       <div>
-        <h2 className="short-bottom">
+        <h2 className="short-top short-bottom">
```

Comment:
Let me handle that difference in the follow-up PR (where we merge the two sections), then